Functions/xASL_test_Flavors.m

```diff
@@ -29,179 +30,84 @@ function xASL_test_Flavors(pathExploreASL, pathTest, bTest, x)
 %              directory.
 %
 % -----------------------------------------------------------------------------------------------------------------------------------------------------
-% EXAMPLE:     xASL_test_Flavors('/home/user/ExploreASL', '/home/user/tmp', [0 1 1 0 1], x)
+% EXAMPLE:     flavors = xASL_test_Flavors(testConfig, [1 0 0 0 0 0 0], x, flavors);
+%
 % __________________________________
 % Copyright 2015-2021 ExploreASL
 
-%% 0. Admin and initialization
-if nargin < 2 || isempty(pathExploreASL) || isempty(pathTest)
-	error('The path to the code and working directory needs to be specified');
-end
-
-if nargin < 3 || isempty(bTest)
-	bTest = ones(1,7);
-end
-
-if length(bTest) < 7
-	bTest(end+1:7) = 1;
-end
-
-% Change directory to ExploreASL root folder
-cd(pathExploreASL);
 
-if nargin < 4 || isempty(x)
-    % Remove existing paths
-    thisDirectory = pwd;
-    if ~isempty(pathExploreASL)
-        xASL_adm_RemoveDirectories(pathExploreASL);
-        cd(pathExploreASL);
+    %% 0. Admin and initialization
+    if isempty(testConfig.pathExploreASL) || isempty(testConfig.pathFlavorDatabase)
+        error('The paths to the code and working directory needs to be specified...');
+    end
+    if nargin < 3 || isempty(bTest)
+        bTest = ones(1,7);
+    end
+    if length(bTest) < 7
+        bTest(end+1:7) = 1;
     end
-    % Initialize ExploreASL
-    ExploreASL_Initialize;
-    cd(thisDirectory);
-end
 
-% Initialize the working paths
-flavorsPath  = fullfile(pathTest, 'FlavorDatabase');
-conversionPath = fullfile(pathTest, 'TmpConversion');
-referencePath = fullfile(pathTest, 'TmpReference');
+    % Change directory to ExploreASL root folder
+    cd(testConfig.pathExploreASL);
+
+    if nargin < 4 || isempty(x)
+        % Remove existing paths
+        thisDirectory = pwd;
+        if ~isempty(testConfig.pathExploreASL)
+            xASL_adm_RemoveDirectories(testConfig.pathExploreASL);
+            cd(testConfig.pathExploreASL);
+        end
+        % Initialize ExploreASL
+        ExploreASL_Initialize;
+        cd(thisDirectory);
+    end
+    
 
-% Default dataPar.json for the testing that is fast to run
-defaultDataPar.x.dataset.subjectRegexp = '^sub-.*$';
-defaultDataPar.x.modules.asl.bUseMNIasDummyStructural = 1; % when no structural data, use ASL-MNI registration
-defaultDataPar.x.settings.Quality = 0; % speed up testing
-defaultDataPar.x.settings.DELETETEMP = 1;
+    %% 1. Remove existing test data
+    if bTest(1)
+        xASL_test_Flavors_RemoveExistingTestData(testConfig);
```

Comment:
If flavorList is not given in test-config, it all crashes. It would be nice - if flavor list is missing - to load by default all flavors.

Reply:
Ahhh okay, I understand. The thing is that this should work if you run the testConfig without flavorList from `xASL_test_FullPipelineTest`, because there it does this: 

```matlab
    % Check for flavor list
    if ~isfield(testConfig,'flavorList')
        testConfig.flavorList = xASL_adm_GetFileList(testConfig.pathFlavorDatabase, [], false, [], true);
    end
```

Reply:
But I can just copy paste that part to `xASL_test_Flavors.m` 👍

Reply:
Check this out 9315553 👍